File: lecture0-30/lecture009/diagonal.java
import java.util.Scanner;
public class diagonal{
    public static Scanner scn = new Scanner (System.in);
    public static void main(String[] args){
         
        System.out.println("enter row an col value ");
        int row = scn.nextInt();
        int col = scn.nextInt();
        System.out.println("enter array ");
        int [][] arr = new int [row][col];
        input(arr);
        System.out.println("diagonal elements are ");
        diagonal(arr);

    }
    public static void input(int [][] arr){
        for(int i=0 ; i<arr.length ; i++){
            for(int j=0 ; j<arr[0].length ; j++){
                arr[i][j] = scn.nextInt();
            }
        }
    }
    public static void diagonal(int [][] arr){
        
        int m=arr[0].length;
        int n=arr.length;
        for(int gap=0 ; gap<m ; gap++){
            for(int i=0 , j=gap ; i<m && j<n ; i++ , j++){
                System.out.print(arr[i][j] + " "); 
            }
        }
        
    }
}

File: lecture009/diagonal.cpp
#include <iostream>
#include <vector>
using namespace std;
void input(vector<vector<int>> &a)
{
    for (int i = 0; i < a.size(); i++)
    {
        for (int j = 0; j < a[0].size(); j++)
        {
            cin >> a[i][j];
        }
    }
}
void print(vector<vector<int>> &a)
{
    for (int i = 0; i < a.size(); i++)
    {
        cout << endl;
        for (int j = 0; j < a[0].size(); j++)
        {
            cout << a[i][j] << " ";
        }
    }
}
void diagonal_(vector<vector<int>> &arr)
{
    int m = arr[0].size();
    int n = arr.size();
    for (int gap = 0; gap < m; gap++)
    {
        for (int i = 0, j = gap; i < n && j < m; i++, j++)
        {
            cout << arr[i][j] << " ";
        }
    }
}
int main(int args, char **argv)
{
    int row, col;
    cout << "enter row and col value " << endl;
    cin >> row >> col;
    vector<vector<int>> arr(row, vector<int>(col, 0));
    cout << "enter array " << endl;
    input(arr);
    cout << "dioganls are " << endl;
    diagonal_(arr);
}